src/main/java/ch/njol/skript/expressions/ExprPermissions.java

```diff
@@ -18,71 +18,135 @@
  */
 package ch.njol.skript.expressions;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.bukkit.permissions.PermissionAttachment;
+import org.bukkit.permissions.PermissionAttachmentInfo;
+import org.eclipse.jdt.annotation.Nullable;
+
 import ch.njol.skript.Skript;
 import ch.njol.skript.classes.Changer.ChangeMode;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
-import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.util.Kleenean;
 import ch.njol.util.coll.CollectionUtils;
 
-import org.bukkit.entity.Player;
-import org.bukkit.event.Event;
-import org.bukkit.permissions.PermissionAttachment;
-import org.bukkit.permissions.PermissionAttachmentInfo;
-import org.eclipse.jdt.annotation.Nullable;
-
-import java.util.HashSet;
-import java.util.Set;
-
-@Name("All Permissions")
-@Description("Returns all permissions of the defined player(s). Note that the modifications to resulting list do not actually change permissions.")
+@Name("Permissions")
+@Description({
+	"All permissions of the provided permissible(s). A permissible is anything that can have permissions (like the player).",
+	"A couple of notes: Plugins like Skript cannot modify the default permissions such as bukkit.* permissions.",
+	"Skript also doesn't save the permissions after server restart.",
+	"You will need a permissions plugin or to save them yourself in variables."
+})
 @Examples("set {_permissions::*} to all permissions of the player")
-@Since("2.2-dev33")
-public class ExprPermissions extends SimpleExpression<String> {
-	
+@Since("2.2-dev33, INSERT VERSION (changers)")
+public class ExprPermissions extends PropertyExpression<Entity, String> {
+
 	static {
-		Skript.registerExpression(ExprPermissions.class, String.class, ExpressionType.PROPERTY, "[(all [[of] the]|the)] permissions (from|of) %players%", "[(all [[of] the]|the)] %players%'[s] permissions");
+		Skript.registerExpression(ExprPermissions.class, String.class, ExpressionType.PROPERTY,
+				"[all [[of] the]|the] permissions (from|of) %entities%",
+				"[all [of] the] %entities%'[s] permissions"
```

Comment:
```suggestion
				"%entities%'[s] permissions"
```
This should just be removed completely